microprofile/messaging/core/src/main/java/io/helidon/microprofile/messaging/MessageContext.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.microprofile.messaging;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ * JVM wide context for every referenced message.
+ */
+public class MessageContext extends ConcurrentHashMap<String, Object> {
```

Comment:
Extending a map is not recommended in general. This is a strange construct - is there a design document that would explain this class and its usage?